libraries/Microsoft.Bot.Builder.BotFramework/BotFrameworkAdapter.cs

```diff
@@ -16,15 +17,18 @@ public class BotFrameworkAdapter : ActivityAdapterBase
     {
         private readonly SimpleCredentialProvider _credentialProvider;
         private readonly MicrosoftAppCredentials _credentials;
+        private readonly HttpClient _httpClient; 
 
-        public BotFrameworkAdapter(IConfiguration configuration) : base()
+        public BotFrameworkAdapter(IConfiguration configuration, HttpClient httpClient = null) : base()
         {
+            _httpClient = httpClient ?? new HttpClient();
             _credentialProvider = new ConfigurationCredentialProvider(configuration);
-            _credentials = new MicrosoftAppCredentials(this._credentialProvider.AppId, _credentialProvider.Password);                       
+            _credentials = new MicrosoftAppCredentials(_credentialProvider.AppId, _credentialProvider.Password);                                   
         }
 
-        public BotFrameworkAdapter(string appId, string appPassword) : base()
+        public BotFrameworkAdapter(string appId, string appPassword, HttpClient httpClient = null) : base()
```

Comment:
> HttpClient httpClient = null [](start = 68, length = 29)

I don't think this is the right way to do this.  If you read the anti-pattern they aren't saying that you should have only one instance ever in your code, but that you should model it as a shared resource as appropriate.
So the signature of these classes could have 1 static instance internal that it uses for requests and it definitely meets the goal of not having the anti-pattern.  The anti-pattern is to create per request.  Having 5 instances across 5 classes seems fine without polluting all of our interfaces.

Reply:
The linked article even shows the static singleton pattern:
```
public class SingleHttpClientInstanceController : ApiController
{
    private static readonly HttpClient HttpClient;

    static SingleHttpClientInstanceController()
    {
        HttpClient = new HttpClient();
    }

    // This method uses the shared instance of HttpClient for every call to GetProductAsync.
    public async Task<Product> GetProductAsync(string id)
    {
        var hostName = HttpContext.Current.Request.Url.Host;
        var result = await HttpClient.GetStringAsync(string.Format("http://{0}:8080/api/...", hostName));
        return new Product { Name = result };
    }
}
```

Reply:
See my response later in the thread.

This approach is far more flexible allowing deva to configure timeouts, connection lifetimes, and even manage and/or add their own default headers if they want. 

This approach has this features, yet remains totally agnostic with respect to a DI framework. 

QnAMaker, and Luis, which send unrelated requests, each have their own which also seems the right choice.